### ISSUE TITLE
Support for Visual Studio project files

### ISSUE THREAD

**Issue by nycdotnet** [TypeStrong/grunt-ts#213]
I've created an npm module called `csproj2ts` that will interrogate a Visual Studio project file (.csproj or .vbproj) for its TypeScript configuration.  I propose updating grunt-ts to use this library to allow using the config inside a project.  Here is a minimalist gruntfile using this proposed functionality:

```javascript
module.exports = function(grunt) {
  grunt.initConfig({
    ts: {
      default : {
        vs: 'myproject.csproj'
      }
    }
  });
  grunt.loadNpmTasks("grunt-ts");
  grunt.registerTask("default", ["ts"]);
};
```

This will allow a very smooth transition to using grunt-ts for users of Visual Studio.  The main benefit is that these users can simply use the Visual Studio TypeScript build properties UI to control how grunt-ts compiles their project.  Plus they get easy access to build configuration items not yet implemented in the Visual Studio UI (such as noEmitOnError).

The integration with Task Runner Explorer extension in VS 2013 should be really nice (see http://www.hanselman.com/blog/IntroducingGulpGruntBowerAndNpmSupportForVisualStudio.aspx).  Integration with Grunt is built-in to VS 2015, so no extension is required.

Another example (rest of file is same as above):

```javascript
  default : {
        vs: {
            project: 'myproject.csproj',
            config: 'Release',  //or 'Debug', etc - the VS config to poll for settings.  Blank uses default.
            ignoreFiles: false, //if you prefer to only use the settings, you could set true
            ignoreSettings: false //if you prefer to only use the files, you could set to true
        }
      }
```

**Details:**
`vs` will not be usable with `files`.
Any settings in `vs` will be overwritten by any corresponding settings specified in the gruntfile; an exception is `src` which will be appended to.

Feedback welcome.  I am working on implementing this now.

**Comment by basarat** [comment]
:+1: you :metal:

**Comment by nycdotnet** [comment]
This is PR #215.  Let's discuss there.  I will need some help.

**Comment by nycdotnet** [comment]
Implemented in 4.0.0 and 4.0.1.
